src/graphql/client.ts

```diff
@@ -5,12 +5,18 @@ import {
   HttpLink,
   from,
 } from '@apollo/client';
+import { getApiTokensFromStorage } from 'hds-react';
 
 import i18n from '../i18n/i18nInit';
-import authService from '../auth/authService';
+import pickProfileApiToken from '../auth/pickProfileApiToken';
+
 const cache = new InMemoryCache();
 const authMiddleware = new ApolloLink((operation, forward) => {
-  const token = authService.getToken();
+  let token = null;
+  const apiTokens = getApiTokensFromStorage();
+  if (apiTokens) {
+    token = pickProfileApiToken(apiTokens);
+  }
```

Comment:
getApiTokensFromStorage(); will return nothing if API tokens are being renewed resulting in an edge case where the user gets an error:

![image](https://github.com/user-attachments/assets/dcce0a2f-4321-4f5d-a840-a315f1b09750)


Or alternatively if the page is refreshed while tokens are not available the request is sent before the tokens are fetched

![image](https://github.com/user-attachments/assets/b997ab9b-d874-49d3-89fe-ccd62fef91b9)

Ideally the client should wait for the tokens (for a set amount of time) and make the request only when tokens are available. Using cached tokens is also not a good idea, because they may be expired (laptops suspending etc) . Not sure if this ApolloLink supports async?

Here's some stuff https://github.com/apollographql/apollo-client/issues/2441

Of course, it could also make sense to try to use the `useGraphQL` in HDS, but that's a bit bigger project in itself, perhaps a separate issue.

Reply:
Found a way to make link of the ApolloClient to wait until the token can be read.

Reply:
Yess, looks good.  :+1: